docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,531 @@
+## simulateTransaction Examples and Tutorials Guide
+
+### Table of Contents
+
+- [simulateTransaction Examples and Tutorials Guide](#simulatetransaction-examples-and-tutorials-guide)
+  - [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [When to Use `simulateTransaction` (whenever you use soroban :D )](#when-to-use-simulatetransaction-whenever-you-use-soroban-d-)
+- [How to Call `simulateTransaction`](#how-to-call-simulatetransaction)
+  - [Using Fetch](#using-fetch)
+  - [Using the JavaScript SDK](#using-the-javascript-sdk)
+- [Understanding the Footprint](#understanding-the-footprint)
+- [Assembling a Transaction](#assembling-a-transaction)
+  - [javascript-sdk-assembleTransaction](#javascript-sdk-assembletransaction)
+- [Handling Archived Ledger Entries](#handling-archived-ledger-entries)
+- [Fees and Resource Usage](#fees-and-resource-usage)
+- [Preflight Error Handling](#preflight-error-handling)
+  - [Common Preflight Errors](#common-preflight-errors)
+- [Backend Code and Workflow](#backend-code-and-workflow)
+- [Further Reading](#further-reading)
+
+## Overview
+
+The `simulateTransaction` endpoint in Soroban RPC allows you to submit a trial contract invocation to simulate how it would be executed by the Stellar network. This simulation calculates the effective transaction data, required authorizations, and minimal resource fee. It provides a way to test and analyze the potential outcomes of a transaction without actually submitting it to the network. It can be a nice way to get contract data as well sometimes.
+
+While calling the method on the rpc server is not the ONLY way to simulate a transaction, it will likely be the most common and easiest way.
+
+Here we will look at the objects involved and their definitions.
+
+**SimulateTransactionParams** is the argument passed to the rpc endpoint for simulateTransaction
+
+```typescript
+interface SimulateTransactionParams {
+  transaction: string; // The Stellar transaction to be simulated, serialized as a base64 string.
+  resourceConfig?: {
+    instructionLeeway: number; // Allow this many extra instructions when budgeting resources.
+  };
+}
+```
+
+**SimulateTransactionResult** is the return result from the call. It includes many useful things! let's take a look.
```

Comment:
imo this should be a reference to the relevant JS file and/or RPC docs to prevent bitrot in the future if the schema changes!

Reply:
I removed it and added a link to where it's laid out in the openrpc files, I had added some custom documentation to the types but i think it's understandable enough if someon reads the link.